cmd/skaffold/app/cmd/config/set.go

```diff
@@ -71,12 +71,16 @@ func setConfigValue(name string, value interface{}) error {
 	fieldType := fieldValue.Type()
 	val := reflect.ValueOf(value)
 
-	if fieldType != val.Type() {
-		return fmt.Errorf("%s is not a valid value for field %s", value, fieldName)
+	if IsZero(val) {
```

Comment:
Are you introducing this to handle empty strings? If that's the case, we can just ignore the empty string case - if user sets "" then the value will be set to "".  
This way we don't have to introduce the  whole `IsZero` function which was decided to be abandoned in the reflect package as well (see [this comment](https://go-review.googlesource.com/c/go/+/23064#message-0d8ad7a065f3b3cc0e368f574d141f9d1d8d4fd9)). For the first branch `value == nil` check should be sufficient to set with reflect.Zero

Reply:
The reason I added this was to make the unset logic generic for all types (i.e. so we can pass `nil` to `setConfigValue` without worrying about the type), in case we added values of different types to the global config (e.g. int, bool, etc) - empty string is actually the only case I didn't need `IsZero` for. However, after thinking about it a little more I think it's fine for now to just pass the empty string when we're unsetting. I'll remove this.